src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java

```diff
@@ -218,20 +269,48 @@ public void processElement(final ProcessContext context) throws Exception {
                             }
 
                             //order by score
-                            final List<PairedEnds> scored = Ordering.natural().reverse().onResultOf((PairedEnds pair) -> pair.score()).immutableSortedCopy(paired.get(true));
+                            List<PairedEnds> scored = Ordering.natural().reverse().onResultOf((PairedEnds pair) -> pair.score()).sortedCopy(paired.get(true));
                             final PairedEnds best = Iterables.getFirst(scored, null);
-                            if (best != null) {
-                                context.output(best.first());
-                                context.output(best.second());
+                            if (best == null) {
+                                return;
+                            }
+                            final String bestName = best.first().getName();
+                            for (PairedEnds pair : scored) {
+                                opticalDuplicateFinder.addLocationInformation(pair.first().getName(), pair);
```

Comment:
Agreed that it's strange, that just seems to be the way the class is designed. If I add it to findOpticalDuplicates I have to change all the other uses. I know that class was heavily optimized so I am a bit wary to change it.